BridgeSDK/BridgeAPI/SBBUploadManager.m

```diff
@@ -377,8 +380,19 @@ - (void)URLSession:(NSURLSession *)session downloadTask:(NSURLSessionDownloadTas
       @"Content-Type": uploadRequest.contentType,
       @"Content-MD5": uploadRequest.contentMd5
       };
+    SBBNetworkManagerTaskCompletionBlock uploadFileCompletion = nil;
```

Comment:
It's a multi-step process to make background upload and confirmation appear as one seamless asynch call with a completion handler. It's actually a series of steps, each triggered by completion of the last. The completion block passed into the uploadFileToBridge:contentType:completion: method is saved, associated with the task ID. The SBBUploadManager is set as the background session delegate for SBBBridgeNetworkManager so the NSURLSession delegate methods are forwarded here. When all the steps are completed (or fail at some point along the way), the original completion block will be looked up and called with the error, if any.